### ISSUE TITLE
Parameter types in method comments sometimes don't match the signature

### ISSUE THREAD

**Issue by jtibshirani** [immutables/immutables#242]
The instance type described in method comments is sometimes inconsistent with the method signature. In this example, the comment references AbstractSearchResult, but the method takes a SearchResult:

```
/**
 * Fill builder with attribute values from provided {@link AbstractSearchResult} instance.
 * Regular attribute values will be replaced with ones of an instance.
 * Instance's absent optional values will not replace present values.
 * @param instance instance to copy values from
 * @return {@code this} builder for chained invocation
*/
public final Builder from(SearchResult instance) {
    ...
}
```
From a quick look at the code, this seems to occur when the generated class has greater visibility than the abstract class, and so is considered the 'primary implementation'.

**Comment by elucash** [comment]
Have to take a look, should be easy to correct. Thank you!

**Comment by jtibshirani** [comment]
No problem, and a big thank you for the great library!

**Comment by elucash** [comment]
released in 2.1.9
